[CI] TranslogTests.testWithRandomException

Found by PR testing but there were other instances in CI:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-darwin-compatibility/482/console

```
07:59:00 org.elasticsearch.index.translog.TranslogTests > testWithRandomException FAILED
07:59:00     java.lang.AssertionError: expected:<3> but was:<0>
07:59:00         at __randomizedtesting.SeedInfo.seed([AD75FA3DC5C69916:CBAD9122BE221707]:0)
07:59:00         at org.junit.Assert.fail(Assert.java:88)
07:59:00         at org.junit.Assert.failNotEquals(Assert.java:834)
07:59:00         at org.junit.Assert.assertEquals(Assert.java:645)
07:59:00         at org.junit.Assert.assertEquals(Assert.java:631)
07:59:00         at org.elasticsearch.index.translog.TranslogTests.testWithRandomException(TranslogTests.java:2726)
07:59:01 REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.translog.TranslogTests.testWithRandomException" -Dtests.seed=AD75FA3DC5C69916 -Dtests.security.manager=true -Dtests.locale=zh-Hans-CN -Dtests.timezone=IET -Dcompiler.java=13
```

--- a/server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java
+++ b/server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java
@@ -2657,9 +2657,9 @@ public class TranslogTests extends ESTestCase {
                             committing = true;
                             failableTLog.getDeletionPolicy().setTranslogGenerationOfLastCommit(failableTLog.currentFileGeneration());
                             failableTLog.getDeletionPolicy().setMinTranslogGenerationForRecovery(failableTLog.currentFileGeneration());
+                            syncedDocs.clear();
                             failableTLog.trimUnreferencedReaders();
                             committing = false;
-                            syncedDocs.clear();
                         }
                     }
                     // we survived all the randomness!!!
@@ -2675,7 +2675,7 @@ public class TranslogTests extends ESTestCase {
                     assertEquals(failableTLog.getTragicException(), ex);
                 } catch (RuntimeException ex) {
                     assertEquals(ex.getMessage(), "simulated");
-                    assertEquals(failableTLog.getTragicException(), ex);
+                    assertNull("Don't consider failures while trimming unreferenced readers as tragedy", failableTLog.getTragicException());
                 } finally {
                     Checkpoint checkpoint = Translog.readCheckpoint(config.getTranslogPath());
                     if (checkpoint.numOps == unsynced.size() + syncedDocs.size()) {

